src/Workspaces/Remote/ServiceHub/Services/CodeAnalysisService_DesignerAttributes.cs

```diff
@@ -17,16 +19,22 @@ internal partial class CodeAnalysisService : IRemoteDesignerAttributeService
         /// 
         /// This will be called by ServiceHub/JsonRpc framework
         /// </summary>
-        public async Task<ImmutableArray<DesignerAttributeDocumentData>> ScanDesignerAttributesAsync(ProjectId projectId, CancellationToken cancellationToken)
+        public async Task<IReadOnlyList<DesignerAttributeDocumentData>> ScanDesignerAttributesAsync(ProjectId projectId, CancellationToken cancellationToken)
         {
             using (RoslynLogger.LogBlock(FunctionId.CodeAnalysisService_GetDesignerAttributesAsync, projectId.DebugName, cancellationToken))
             {
                 var solution = await GetSolutionAsync(cancellationToken).ConfigureAwait(false);
                 var project = solution.GetProject(projectId);
+
                 var data = await AbstractDesignerAttributeService.TryAnalyzeProjectInCurrentProcessAsync(
                     project, cancellationToken).ConfigureAwait(false);
 
-                return data.Values.ToImmutableArray();
+                if (data.Count == 0)
+                {
+                    return SpecializedCollections.EmptyReadOnlyList<DesignerAttributeDocumentData>();
+                }
+
+                return data.Values.ToList();
```

Comment:
I dont believe that's how it works.
https://github.com/JamesNK/Newtonsoft.Json/blob/b65807bbd2be51e8fa5bfebb50af5786b3eeb9ed/Src/Newtonsoft.Json/Serialization/JsonArrayContract.cs#L181

it will just create ReadOnlyCollection